salt/metalk8s/orchestrate/deploy_new_node.sls

```diff
@@ -0,0 +1,65 @@
+{%- if pillar['bootstrap_id'] %}
+{%-   set control_plane_ips = salt.saltutil.runner('mine.get', tgt=pillar['bootstrap_id'], fun='control_plane_ip') %}
+{%- else %}
+{%-   set control_plane_ips = {} %}
+{%- endif %}
+
+{%- if pillar['bootstrap_id'] in control_plane_ips.keys() and control_plane_ips[pillar['bootstrap_id']] %}
+{%-   set control_plane_ip = control_plane_ips[pillar['bootstrap_id']] %}
+{%- else %}
+{%-   set control_plane_ip = 'localhost' %}
+{%- endif %}
+
+
+Deploy salt-minion on a new node:
+  salt.state:
+    - ssh: true
+    - roster: kubernetes
+    - tgt: {{ pillar['node_name'] }}
+    - saltenv: {{ saltenv }}
+    - sls:
+      - metalk8s.roles.salt-minion
+    - pillar:
+        repo:
+          host: {{ control_plane_ip }}
+        salt:
+          master:
+            host: {{ control_plane_ip }}
+
+Set grains:
+  salt.state:
+    - tgt: {{ pillar['node_name'] }}
+    - sls:
+      - metalk8s.node.grains
+    - require:
+      - salt: Deploy salt-minion on a new node
+
+Refresh the mine:
+  salt.function:
+    - name: mine.update
+    - tgt: '*'
```

Comment:
Nope, only that node.

Reply:
Hmm no we need to have all the etcd endpoint so we should refresh all minions mine.

Reply:
Won't this fail, then, when one of the minions is unavailable?

Reply:
Hmm maybe but i think until we move it away from the mine we should fail if we cant update the mine as we will maybe create a wrong etcd node after